FFmpegInterop/FFmpegInteropMSS.cpp

```diff
@@ -296,6 +297,31 @@ MediaPlaybackItem^ FFmpegInteropMSS::CreateMediaPlaybackItem(TimeSpan startTime,
 	}
 }
 
+IAsyncAction^ FFmpegInterop::FFmpegInteropMSS::ParseExternalSubtitleStream(IRandomAccessStream ^ stream)
+{
+	return create_async([this, stream]
+	{
+		auto config = ref new FFmpegInteropConfig();
+		config->IsExternalSubtitleParser = true;
+
+		auto externalSubsParser = FFmpegInteropMSS::createFromStreamInternal(stream, config, nullptr);
+		int readResult = 0;
+		while ((readResult = externalSubsParser->m_pReader->ReadPacket()) >= 0)
+		{
+			Concurrency::interruption_point();
+		}
+		for each(auto externalSubtitle in externalSubsParser->SubtitleStreams)
+		{
+			subtitleStrInfos->Append(externalSubtitle);
+			if (this->PlaybackItem != nullptr)
+			{
+				PlaybackItem->Source->ExternalTimedMetadataTracks->Append(externalSubtitle->SubtitleTrack);
```

Comment:
I think we might need some kind of "Detach" method on SubtitleProvider class, which will detach event handlers and set SubtitleTrack to nullptr. Check SubtitleProvider.Initialize() and dtor. This would then be called after we attach the TimedMetadataTrack to MediaPlaybackItem, to ensure that any references to old interop class are released.

Reply:
Not 100% sure where/how it needs to be done, be we should make sure that external interop stuff gets released after end of this method...

Reply:
I think an explicit call to delete + detaching the event handlers should do it. I had to reinstall my work machine, so no joy today.

Reply:
Oh that sucks, good luck with re-installing...

Reply:
Ok cool